src/qibolab/pulses/sequence.py

```diff
@@ -34,8 +34,13 @@ def channel_duration(self, channel: str) -> float:
         """Duration of the given channel."""
         return sum(item.duration for item in self[channel])
 
-    def __add__(self, other: "PulseSequence") -> "PulseSequence":
-        """Create a PulseSequence which is self + necessary delays to
-        synchronize channels + other."""
-        # TODO: implement
-        ...
+    def append(self, other: "PulseSequence") -> None:
+        """Appends other in-place such that the result is self + necessary
+        delays to synchronize channels + other."""
+        tol = 1e-12
+        durations = {ch: self.channel_duration(ch) for ch in other}
+        max_duration = max(durations.values(), default=0.0)
+        for ch, duration in durations.items():
+            if delay := round(max_duration - duration, int(1 / tol)) > 0:
```

Comment:
I like that is very concise, but side-effects are a bit implicit, if possible I'd avoid.

Moreover, unless you are very familiar with `:=`, the priority wrt `>` is not incredibly clear.

I'd just go for something much more basic, and for sure more verbose, but completely equivalent:
```suggestion
            delay = round(max_duration - duration, int(1 / tol))
            if delay > 0:
```